### ISSUE TITLE
select-to-quote drops formatting and doesn't escape HTML

### ISSUE THREAD

**Issue by yamikuronue** [apxltd/what-bugs#6]
As shown in this thread: https://what.thedailywtf.com/topic/19205/blakeyrat-pointing-out-nodebb-problems/159

**Comment by sjennison** [comment]
Suspect a bug with @BenLubar's sanitizer, norepro on my local forum

![image](https://cloud.githubusercontent.com/assets/5776850/13941667/63478c86-efc0-11e5-8efa-54fb2b05b88e.png)

**Comment by BenLubar** [comment]
@sjennison NodeBB doesn't support HTML by default, and this is what happens when you open an `<a>` and don't close it.

It's a bug with select-to-quote if anything.

**Comment by sjennison** [comment]
Isn't your HTML sanitizer supposed to fix that though? :stuck_out_tongue:

**Comment by sjennison** [comment]
Or... well, let me try a repro again

**Comment by sjennison** [comment]
Yeah, it's some interaction between html, select-to-quote, and probably your sanitizer (shouldn't it close that `<a>`?)

**Comment by BenLubar** [comment]
Try this: `data:text/html,<blockquote>some quote with <a href> link</blockquote>some more text`

It looks like my sanitizer is following the HTML spec.

**Comment by sjennison** [comment]
Well what do you mean by following the spec? I mean, it preserves the broken tag, but is that the intent?

I suppose I can see a reason for that behavior though... Although an unclosed nested `<a>` is against the spec anyways...

![image](https://cloud.githubusercontent.com/assets/5776850/13942002/eb121f16-efc3-11e5-9c6a-2211d0db14e3.png)

So I guess the question becomes "Is it in your sanitizer's job description to fix broken HTML?"
